client_cli/src/main.rs

```diff
@@ -1118,334 +1118,91 @@ mod json {
         }
     }
 }
-
 #[cfg(test)]
 mod tests {
-    use iroha_client::{
-        config::{Configuration as ClientConfiguration, ConfigurationProxy},
-        data_model::{
-            asset::{AssetDefinitionId, AssetId},
-            domain::DomainId,
-            name::Name,
-        },
-    };
+    use std::str::FromStr;
+
+    use serde_json::Value as JsonValue;
 
     use super::*;
-    use crate::asset::{RemoveKeyValue, SetKeyValue};
 
-    struct TestContext {
-        config: ClientConfiguration,
-        skip_mst_check: bool,
+    #[test]
+    fn parse_ipv4_address() {
+        match ValueArg::from_str("192.168.0.1") {
+            Ok(ValueArg(Value::Ipv4Addr(addr))) => {
+                assert_eq!(addr, Ipv4Addr::new([192, 168, 0, 1]))
+            }
+            _ => panic!("Expected Ipv4Addr parsing to succeed."),
+        }
```

Comment:
Could be written in a flat way, with less code and with detailed error message:

```rs
let actual = ValueArg::from_str("192.168.0.1").expect("Should not fail with valid IP input");
assert_eq!(actual, ValueArg(Value::Ipv4Addr(Ipv4Addr::new([192, 168, 0, 1]))))
```

Also consider [`assert_matches!`](https://docs.rs/assert_matches/latest/assert_matches/macro.assert_matches.html) (not an `std`'s unstable one).